run.py

```diff
@@ -156,6 +158,228 @@ def reindex(since=None, batch_size=None, force=False, update_solr=True, update_m
+              'author_count', 'author_facet', 'author_facet_hier', 'author_norm', 'bibgroup', 'bibgroup_facet',
+              'bibstem', 'bibstem_facet', 'body', 'citation', 'citation_count', 'cite_read_boost', 'classic_factor',
+              'comment', 'copyright', 'data', 'data_count', 'data_facet', 'database', 'date', 'doctype', 'doctype_facet_hier',
+              'doi', 'eid', 'email', 'entry_date', 'esources', 'first_author', 'first_author_facet_hier',
+              'first_author_norm', 'fulltext_mtime', 'grant', 'grant_facet_hier', 'id', 'identifier', 'indexstamp',
+              'isbn', 'issn', 'issue', 'keyword','keyword_facet', 'keyword_norm', 'keyword_schema', 'lang',
+              'links_data', 'metadata_mtime', 'metrics_mtime', 'nedid', 'nedtype', 'ned_object_facet_hier', 'nonbib_mtime',
+              'origin', 'orcid_mtime', 'orcid', 'orcid_pub', 'orcid_user', 'orcid_other', 'page', 'page_range',
+              'page_count', 'property', 'pub', 'pub_raw', 'pubdate', 'pubnote', 'read_count', 'reader', 'recid',
+              'reference', 'simbad_object_facet_hier', 'simbid', 'simbtype', 'title', 'update_timestamp', 'vizier',
+              'vizier_facet', 'volume', 'year')
+
+    ignore_fields = ('id', 'indexstamp', 'fulltext_mtime', 'links_data', 'metadata_mtime', 'metrics_mtime',
+                     'nonbib_mtime', 'orcid_mtime', 'recid', 'update_timestamp')
+
+    mismatch = 0
+    missing_required = 0
+    missing = 0
+    notins1 = 0
+    notins2 = 0
+
+    for field in fields:
+        if field not in ignore_fields:
+            match = fields_match(bibcode, s1, s2, field, validation_logger)
+            if match != True:
+                if match == False:
+                    mismatch += 1
+                if match == 'required new field not in bibcode':
+                    missing_required += 1
+                    missing += 1
+                if match == 'field not in bibcode':
+                    missing += 1
+                if match == 'field not in s1':
+                    notins1 += 1
+                if match == 'field not in s2':
+                    notins2 += 1
+
+    if mismatch == 0:
+        validation_logger.info('Bibcode {}: no mismatched fields'.format(bibcode))
+
+    validation_logger.info('The following fields are ignored: {}'.format(ignore_fields))
+    validation_logger.info('Mismatch stats for bibcode {}: {} mismatches, {} missing required new fields, '
+                           '{} fields not in old database, {} fields not in new database, '
+                           '{} fields not in either database'.format(bibcode,mismatch,missing_required,notins1,notins2,missing))
+
+
+def fields_match(bibcode, s1, s2, field, validation_logger):
+    new_fields = ('data_count', 'entry_date', 'esources', 'nedid', 'nedtype', 'ned_object_facet_hier', 'origin',
+                  'page_count', 'page_range')
+
+    if (field in s1) & (field in s2):
```

Comment:
:-) cool! but might be safer to use && (that's just for the future generations that don't understand how bitmasks work)

and to be fair, it is easy to miss '&'